homeassistant/components/netgear_lte/__init__.py

```diff
@@ -137,62 +148,83 @@ def get_modem_data(self, config):
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up Netgear LTE component."""
-    if DOMAIN not in hass.data:
-        websession = async_create_clientsession(
-            hass, cookie_jar=aiohttp.CookieJar(unsafe=True)
+    hass.data[DATA_HASS_CONFIG] = config
+
+    if lte_config := config.get(DOMAIN):
+        for entry in lte_config:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=entry
+                )
+            )
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.6.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "Netgear LTE",
+            },
         )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Netgear LTE from a config entry."""
+    host = entry.data[CONF_HOST]
+    password = entry.data[CONF_PASSWORD]
+
+    if DOMAIN not in hass.data:
+        websession = async_create_clientsession(hass, cookie_jar=CookieJar(unsafe=True))
+
         hass.data[DOMAIN] = LTEData(websession)
 
-        await async_setup_services(hass)
+    modem = Modem(
+        hostname=host,
+        password=password,
+        websession=hass.data[DOMAIN].websession,
+    )
+    try:
+        await modem.login()
+    except Error as ex:
+        raise ConfigEntryNotReady("Cannot connect/authenticate") from ex
 
-    netgear_lte_config = config[DOMAIN]
+    # Set up modem
+    await _setup_lte(hass, entry.data)
```

Comment:
We already the modem above. I don't think we should do it twice. We also need to store the modem somewhere, right?

Reply:
The modem is stored in the `ModemData` object. This integration can benefit from a coordinator in a future PR.

Reply:
Yes, that modem instance is stored but not the modem instance we create a couple of lines above. That's my point. We shouldn't create two different modem instances and the one we create should be stored. We need to refactor this.